pkg/promtail/promtail_test.go

```diff
@@ -390,11 +465,15 @@ func (h *testServerHandler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 			h.t.Error("Expected to find a label with name `filename` but did not!")
 			return
 		}
+
 		if _, ok := h.receivedMap[file]; ok {
 			h.receivedMap[file] = append(h.receivedMap[file], s.Entries...)
 		} else {
 			h.receivedMap[file] = s.Entries
 		}
+
+		h.receivedLabels[file] = labels
```

Comment:
the only problem with this approach is that labels can be different for each log line processed, this will effectively only hold the labels from the last log line processed.

You could make `receivedLabels map[string]labels.Labels` -> `receivedLabels map[string][]labels.Labels` and append the labels from each log line to the list?

I would change the labels for one of your two log lines (maybe change the action from GET to POST) and then you can confirm each log line is getting the proper labels